modules/jawn/src/main/scala/io/circe/jawn/JawnParser.scala

```diff
@@ -18,16 +18,37 @@ final object JawnParser {
    * input, and decoding a ten-million digit JSON number into a `BigInteger` may
    * take minutes.
    */
-  def apply(maxValueSize: Int): JawnParser = new JawnParser(Some(maxValueSize))
+  def apply(maxValueSize: Int): JawnParser = new JawnParser(Some(maxValueSize), allowDuplicateKeys = true)
+
+  /**
+    * Returns a parser that fails on:
+    * <ul>
+    *   <li>JSON strings, object keys, or numbers that exceed a given length </li>
+    *   <li>encountering duplicate keys as per JSONlint</li>
+    * </ul>
+    *
+    * In some cases excessively long values (e.g. JSON numbers with millions of
+    * digits) may support denial-of-service attacks. For example, the string
+    * constructor for Java's `BigInteger` is quadratic with the length of the
+    * input, and decoding a ten-million digit JSON number into a `BigInteger` may
+    * take minutes.
+    */
+  def apply(maxValueSize: Int, allowDuplicateKeys: Boolean): JawnParser = new JawnParser(Some(maxValueSize), allowDuplicateKeys)
+
+
+  /**
+    * Returns a parser that fails on encountering duplicate keys as per JSONlint.
```

Comment:
This comment doesn't seem accurate, since the user could provide `allowDuplicateKeys = true`.

Reply:
I think I'd prefer to omit this overload.